### PR TITLE
CLOUD-2091: Changes for SPA app

### PR DESCRIPTION
**Pull request by sdoguparti**
# Description

Please look into the ticket mentioned below for details

Fixes [#2091](https://usxpress.atlassian.net/browse/CLOUD-2091)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Tested this locally by providing the required variables in terraform.tfvars.json. Please look at the screenshots provided in the ticket.

- [x] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by stefarnold — DISMISSED** [review]
lgtm

File: main.tf [hunk 1]
```diff
@@ -158,15 +168,19 @@ resource "azuread_application" "main_app" {
...
-          id   = data.azuread_service_principal.service_apps[required_resource_access.key].app_role_ids[resource_access.key]
+          id   = data.azuread_application.service_apps[required_resource_access.key].app_role_ids[resource_access.key]
           type = "Role"
         }
       }
     }
   }
 
+  single_page_application {
+    redirect_uris = var.type == "spa" ? var.redirect_uris : []
```
Comment (zumic96) [inline main.tf:179]: In readme it says SPA or API which is upper case and here you check for lower I would suggest change in readme and/or add lower function to this check lower(var.type)
same thing for redirect_uris in web block

File: outputs.tf
Comment (zumic96) [inline outputs.tf:1]: Do we need this outputs for anything as secret will be mounted to pod anyway?

File: outputs.tf [hunk 2]
```diff
@@ -20,10 +20,10 @@ output "client_id" {
 
 output "app_aws_secrets_arn" {
```
Comment (zumic96) [inline outputs.tf:21]: I think this output is not needed, as in terraform-variant-apps we just access the values of other one.

File: main.tf (hunk 1 above)
Comment (sdoguparti) [reply main.tf:179]: Changed the description to lowercase spa and api to avoid confusion.

File: outputs.tf (hunk 2 above)
Comment (sdoguparti) [reply outputs.tf]: Removed this from outputs

**Review by zumic96 — APPROVED** [review]

**Pull request merged**